core/templates/components/forms/schema-based-editors/schema-based-html-editor.directive.ts

```diff
@@ -32,6 +32,15 @@ angular.module('oppia').directive('schemaBasedHtmlEditor', [
       },
       template: require('./schema-based-html-editor.directive.html'),
       controllerAs: '$ctrl',
-      controller: [function() {}]
+      controller: ['$scope', function($scope) {
+        var ctrl = this;
+        ctrl.updateValue = function(value: string) {
+          ctrl.localValue = value;
+          $scope.$applyAsync();
+          setTimeout(() => {
+            $scope.$applyAsync();
+          });
```

Comment:
Why do we need this (in addition to the `$scope.$applyAsync()` in L31)?

Reply:
The first $scope.$applyAsync caused propagated the updated values. The second one is to run the checks on the updated value (if any).